Doc/c-api/list.rst

```diff
@@ -51,6 +51,23 @@ List Objects
    ``len(list)`` on a list object.
 
 
+.. c:function:: int PyList_Clear(PyObject *list)
+
+   Remove all items from *list*. Similar to:
+   ``PyList_SetSlice(list, 0, PY_SSIZE_T_MAX, NULL)``.
+
+   Raise an exception and return ``-1`` if *list* is not a :class:`list`
+   object.  Return 0 on success.
+
+   .. note::
+
+      If the finalizer (:meth:`object.__del__`) of a deleted item modify the
```

Comment:
Is it a useful note for the reader? This is a particularly unlikely situation, unless you're deliberately provoking it.